cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/preview/DefaultPreviewStore.java

```diff
@@ -202,4 +233,65 @@ void clear() throws IOException {
     service.dropTable(PREVIEW_TABLE_ID.getDataset());
     service.ensureTableExists(PREVIEW_TABLE_ID.getDataset());
   }
+
+  @Override
+  public PreviewJobQueueState add(PreviewJob previewJob) {
+    // PreviewStore is a singleton and we have to create gson for each operation since gson is not thread safe.
```

Comment:
Gson is not thread safe?? We've been using it as static final in a log of classes.